src/io/iter_prefetcher.h

```diff
@@ -79,17 +81,22 @@ class PrefetcherIter : public IIterator<DataBatch> {
           (*dptr)->data.resize(batch.data.size());
           (*dptr)->index.resize(batch.batch_size);
           for (size_t i = 0; i < batch.data.size(); ++i) {
+            auto dtype = param_.dtype != INT_MAX
```

Comment:
@gengyifeng Why was this logic here? The change of default value broke some downstream iterators